### ISSUE TITLE
Synchronize saved tabs across Chrome instances

### ISSUE THREAD

**Issue by peterjmag** [tabalanche/tabalanche-extension#26]
I'm not terribly familiar with PouchDB, but it appears to be built for this kind of thing. (And perhaps you've already implemented it—I can't test it myself at the moment, so apologies if you have!)

Could also be possible with [`chrome.storage`](http://stackoverflow.com/a/4904316/349353), which you appear to be using for syncing settings already. 

One more idea: it would be great to save the "source" (e.g. the hostname if possible, or some other computer-specific identifier) of each tab group when syncing across multiple computers. That way I could easily distinguish between tab groups I saved at work vs. those I saved at home (as an example), and maybe even use it as a filter in the dashboard at some point in the future.

Thanks for all your work on this project so far! Looking forward to using it and contributing.

**Comment by stuartpb** [comment]
> I'm not terribly familiar with PouchDB, but it appears to be built for this kind of thing. (And perhaps you've already implemented it—I can't test it myself at the moment, so apologies if you have!)

Yep - haven't implemented it yet, but it's one of the first things I plan to implement now that I've got a serviceable development workflow with [Pull Tool](https://github.com/pulltool/pulltool-chrome-app).

> Could also be possible with `chrome.storage`, which you appear to be using for syncing settings already.

The thing about `chrome.storage` is that extensions only have a limited amount of space to use it (and it's significantly less than what's made available via IndexedDB), so I'm only using it for fixed-size data like, as you noted, settings.

> One more idea: it would be great to save the "source" (e.g. the hostname if possible, or some other computer-specific identifier) of each tab group when syncing across multiple computers. That way I could easily distinguish between tab groups I saved at work vs. those I saved at home (as an example), and maybe even use it as a filter in the dashboard at some point in the future.

I guess I can hack this with [`chrome.sessions.getDevices`](https://developer.chrome.com/extensions/sessions#method-getDevices) - this'll be a lot easier after the [`chrome.signedInDevices`](https://developer.chrome.com/extensions/signedInDevices) API goes live (if it ever does). Might be worth tracking https://code.google.com/p/chromium/issues/detail?id=459248.

There's also a planned feature for categorization down the line that should help with this.

**Comment by stuartpb** [comment]
Classifying as 3floc (~100-999 lines of code to implement) based on expected auth complexity.

**Comment by alan23273850** [comment]
Has this feature been already implemented yet? I'm so excited about it!
